### ISSUE TITLE
Redis input plugin: support free form of queries

### ISSUE THREAD

**Issue by bitxel** [influxdata/telegraf#3200]
## Feature Request

Measure redis list length

### Proposal:

In many cases, we use redis list type as a simple message queue. Redis support the magic atomic command `RPUSHLPOP`, so that we can safely consume each message only once.

### Current behavior:

Currently, telegraf has a input plugin `redis`, which can measure and structure the response of  `INFO` command. It's useful for us to have an overview of current status of redis service.

### Desired behavior:

Measure the length of specific keys(queues). So that we can know the status of each queue. It's useful for us to adjust the number of worker processes and it's useful to find out the issue of our systems.

I'd like to know if it accords with the design principle of telegraf, and if anyone has the similar need. After checked the inputs plugins, most of which are system level or service level measurements. It's actually an application level measurement. 

We have build a `redismq` input plugin based on all master's work. If it's ok to add it in inputs plugin, I'll submit a PR:)  

Thanks

**Comment by danielnelson** [comment]
I think this is too specific to your use case for inclusion as is.  A plugin that allowed for free form redis queries would be the closest thing that would fit.

**Comment by bitxel** [comment]
yep, good idea, extend `redis` input plugin to support more redis queries except for `INFO`.

**Comment by idanci** [comment]
Looks like a duplicate of https://github.com/influxdata/telegraf/issues/1027

**Comment by danielnelson** [comment]
@idanci Good catch, thanks for letting me know.
